### ISSUE TITLE
Incorrect decoding of file names with some zip files

### ISSUE THREAD

**Issue by KenobiTom** [srikanth-lingala/zip4j#432]
Due to the changes introduced with [#403](https://github.com/srikanth-lingala/zip4j/issues/403) zip files created with Windows and extracted via zip4j are no longer resulting in correct file names. The fix use of UTF-8 in this context destroys "Umlaute" (öäü...).
For example täglich.txt becomes t�glich.txt. 

This behavior indicates that the original approach to fallback on the zip spec. encoding (cp437) rather than UTF-8 if there is no encoding supplied by the packing tool seemed to be the better one in this case.

A solution for this could be to enable the user to choose the behavior of zip4j: fallback to zip spec or UTF-8 if there is no encoding supplied.

The corresponding change is here:
https://github.com/srikanth-lingala/zip4j/commit/d80df16c0479776a65d0a290d59dad54c9bef540

return new String(data, ZIP_STANDARD_CHARSET_NAME);  // Cp437
vs.
return new String(data, ZIP4J_DEFAULT_CHARSET); // UTF-8

Test zip containing 1 file:
[tester.zip](https://github.com/srikanth-lingala/zip4j/files/8830541/tester.zip)

Code snippet:

ZipFile zipFile = new ZipFile(source);

// Get a list of FileHeaders. 
List<FileHeader> fileHeaderList = zipFile.getFileHeaders();
// Loop through all the fileHeaders
for (FileHeader fileHeader : fileHeaderList) {
	if (fileHeader != null) {
        // Build the output file
        File outFile = new File(destination, fileHeader.getFileName());
	//...
}

**Comment by srikanth-lingala** [comment]
Thanks for the detailed analysis. I agree with you that a fallback to zip standard charset (Cp437) is better than utf-8. Zip specification defines this behaviour as well, so this will be in-line with the specification. Some other libraries (apache commons compress, for example) use utf-8 by default, which although makes sense to some extent (because most tools use utf8 now-a-days), this is not in sync with the zip specification and will fail in cases like the one you saw here.

I reverted the change I did with the [previous commit][1]. If clients want to force zip4j to use utf8 they can set it via `ZipFile.setCharset(StandardCharsets.UTF-8)`.

[1]: https://github.com/srikanth-lingala/zip4j/commit/d80df16c0479776a65d0a290d59dad54c9bef540
